multicast/hear.py

```diff
@@ -259,48 +262,15 @@
 			def kill_func(a_server):
 				if a_server is not None:
 					a_server.shutdown()
-			end_thread = _threading.Thread(name="Kill_Thread", target=kill_func, args=[self])
+			end_thread = threading.Thread(name="Kill_Thread", target=kill_func, args=[self])
 			end_thread.start()
-		super(McastServer, self).handle_error(request, client_address)
-
-
-class MyUDPHandler(socketserver.BaseRequestHandler):
-	"""Subclasses socketserver.BaseRequestHandler for handling echo function.
-
-	Basically simplifies testing by allowing a trivial echo back (case-insensitive) of string
-	data, after printing the sender's ip out.
-
-	Minimal Acceptance Testing:
-
-	First set up test fixtures by importing multicast.
-
-	Testcase 0: Multicast should be importable.
-
-		>>> import multicast
-		>>> multicast.hear is not None
-		True
-		>>> from multicast.hear import MyUDPHandler as MyUDPHandler
-		>>>
-
-	Testcase 1: MyUDPHandler should be automatically imported.
-
-		>>> MyUDPHandler.__name__ is not None
-		True
-		>>>
-
-
-	"""
-
-	def handle(self):
-		data = self.request[0].strip()
-		socket = self.request[1]
-		print(str("{} wrote:").format(self.client_address[0]))
-		print(data)
-		socket.sendto(data.upper(), self.client_address)
+		else:
+			super(McastServer, self).handle_error(request, client_address)
```

Comment:
what would be a safer non-blocking mechanism? do elaborate.

Reply:
🤔 can we move the actual `self.shutdown()` to the already overloaded `open_request()` function [around line 146](https://github.com/reactive-firewall/multicast/pull/148/files/9dd60e3068338f61ed5d7aaee0d422684fe83432#diff-90aabb1483b333cc5349632d3bf9ee9dc0a4b43c4223159838522a7c985e58e7R246), instead of altering the `serve_forever` function? Additionally can we implement a `should_shutdown` accessor for further subclasses to possibly customize and use that for our check (simplifying future maintenance), and then consider the `should_shutdown` a private instance variable?

Reply:
🗒️ 1. Please, make a detailed comment including the improved implementation on #134 for posterity.

😕 2. Please break down exactly what the `@property` decorator does, and what does the official documentation say about it? and where can I find the official documentation? and When was the `@property` introduced to python? and where can I find that notice officially documented? Does it remain consistent across all python 3 versions?